exonum/src/storage/proof_map_index/proof.rs

```diff
@@ -653,13 +642,48 @@ where
         });
 
         collect(&proof).map(|h| {
-            (
-                entries
-                    .into_iter()
-                    .filter_map(OptionalEntry::into_kv)
-                    .collect(),
-                h,
-            )
+            CheckedMapProof {
+                entries: entries.into_iter().map(OptionalEntry::into).collect(),
+                hash: h,
+            }
         })
     }
 }
+
+impl<K, V> CheckedMapProof<K, V> {
+    /// Retrieves references to keys that the proof shows as missing from the map.
+    pub fn missing_keys(&self) -> Vec<&K> {
```

Comment:
I would like to want to return `impl Iterator<Item = &K>` here, but `impl Traits` are not stable yet.